### PR TITLE
Impact sound logic for melee Hits

### PR DESCRIPTION
**Pull request by Z3rca**
warning:  requires Impact sound branch of the environment Module

it just adds impact sounds for the different damage types. In a 30 meter distance, sounds will not be played to save performance. While the logic seems to work very well, the sounds themselves are less good perceivable.
I would like to merge those, to give Duke the opportunity to fine-tune the impact sounds. 

I also made a  refractor for voices, since I believe, that the folder group could contain more custom sound logic, rather than just voices.

### CODE REVIEW

File: CSharpSourceCode/TOW_Core.csproj [hunk 1]
```diff
@@ -16,7 +16,7 @@
   </PropertyGroup>
   <!-- Bannerlord path Property-->
   <PropertyGroup>
-    <GameDir>C:\Program Files (x86)\Steam\steamapps\common\Mount &amp; Blade II Bannerlord</GameDir>
+    <GameDir>G:\Program Files (x86)\SteamLibrary\steamapps\common\Mount &amp; Blade II Bannerlord</GameDir>
```
Comment (hunharibo) [inline CSharpSourceCode/TOW_Core.csproj]: don't commit this pls :D :D :D

**Review by hunharibo — APPROVED** [review]
looks good other than the local exe path :)

File: CSharpSourceCode/TOW_Core.csproj (hunk 1 above)
Comment (Z3rca) [reply CSharpSourceCode/TOW_Core.csproj]: -.- i will never get this straight

Comment (Z3rca) [reply CSharpSourceCode/TOW_Core.csproj]: fixed

**Pull request merged**